plugins/image-prioritizer/class-ip-img-tag-visitor.php

```diff
@@ -72,8 +72,41 @@ public function __invoke( OD_HTML_Tag_Walker $walker ): bool {
 			$walker->remove_attribute( 'fetchpriority' );
 		}
 
-		// TODO: If the image is visible (intersectionRatio!=0) in any of the URL metrics, remove loading=lazy.
-		// TODO: Conversely, if an image is the LCP element for one breakpoint but not another, add loading=lazy. This won't hurt performance since the image is being preloaded.
+		// TODO: Also if the element isLCPCandidate it should never by lazy-loaded.
+		$is_visible = false;
+		$is_found   = false;
+		foreach ( $this->url_metrics_group_collection as $group ) {
```

Comment:
There should be a `OD_URL_Metrics_Group_Collection::get_element_by_xpath()` method which optimizes and caches this lookup.